eclipse-theia/theia
# Ensure preference files only in problems view if open

#### What it does

Fixes #7154 by disposing of the preference system's reference to the `MonacoEditorModel` when not in use. This is similar to [VSCode's approach](https://github.com/microsoft/vscode/blob/edbf56a72a0630649effa940940d0800d42753e2/src/vs/workbench/services/configuration/common/configurationEditingService.ts#L184) in which the model reference is disposed of at the end of the write.

#### How to test

1. Open the application
2. Open the problems view
3. Open a preference file
4. Ensure that the preference file has some problems (dangling commas, schema violations)
5. Close the file
6. Observe that the problems view does not continue to show the problems for the preference file when it is closed.
---
1. Do normal preference operations (set values in the UI, set values in the file)
2. Observe that the changes take effect as expected.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
